activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -7,17 +7,27 @@ def collection_cache_key(collection = all, timestamp_column = :updated_at) # :no
       if collection.loaded?
         size = collection.size
         if size > 0
-          timestamp = collection.max_by(&timestamp_column).public_send(timestamp_column)
+          timestamp = collection.max_by(&timestamp_column)._read_attribute(timestamp_column)
```

Comment:
This enforces a requirement that the timestamp column is an actual attribute and doesn't give the developer an opportunity to override it's value by redefining the attribute getter. Seems like any performance gain would be negligible and the `max_by` is calling the method on every record anyway.